src/app/core/services/grouping/grouping-context.service.ts

```diff
@@ -36,12 +38,36 @@ export class GroupingContextService {
   }
 
   /**
-   * Sets the current grouping type.
-   * @param groupBy - The grouping type to set.
+   * Initializes the service from URL parameters.
    */
-  setCurrentGroupingType(groupBy: GroupBy): void {
+  initializeFromUrlParams() {
+    const groupByParam = this.route.snapshot.queryParamMap.get(GroupingContextService.GROUP_BY_QUERY_PARAM_KEY);
+
+    if (groupByParam && Object.values(GroupBy).includes(groupByParam as GroupBy)) {
+      this.setCurrentGroupingType(groupByParam as GroupBy, true);
+    } else {
+      this.setCurrentGroupingType(DEFAULT_GROUPBY, true);
+    }
+  }
+
+  /**
+   * Sets the current grouping type and updates the corresponding query parameter in the URL.
+   * @param groupBy The grouping type to set.
+   * @param replaceUrl Determines whether to replace the current URL in the browser's history.
+   *                   If true, it replaces the URL; if false, it adds a new entry to the history.
+   */
+  setCurrentGroupingType(groupBy: GroupBy, replaceUrl: boolean): void {
```

Comment:
Is there a necessity for specifying replaceURL for group by?
In my opinion, it is unnatural for me to be changing back to prior group by if I press the previous page button on my browser, I would instead expect to go to a previous page.
It is a matter of UX and should be discussed further, if others agree then we should always set replaceURL to true and remove this parameter to prevent polluting the user history with unintuitive routes specifying different group bys.

Reply:
Just fixed it to always set 'replaceUrl' to true.